src/vs/workbench/browser/parts/sidebar/sidebarPart.ts

```diff
@@ -106,7 +106,8 @@ export class SidebarPart extends CompositePart<PaneComposite> implements IPaneCo
 			themeService,
 			Registry.as<PaneCompositeRegistry>(ViewletExtensions.Viewlets),
 			SidebarPart.activeViewletSettingsKey,
-			viewDescriptorService.getDefaultViewContainer(ViewContainerLocation.Sidebar)!.id,
+			// During unit testing, it may be empty and needs.
+			viewDescriptorService.getDefaultViewContainer(ViewContainerLocation.Sidebar)?.id || '',
```

Comment:
Why is this needed? Are any tests currently failing because of this?

Reply:
> Why is this needed? Are any tests currently failing because of this?

Now the existing unit tests in VS Code are not executed into the SidebarPart constructor, so this writing (use `!` rather than `?` ) does not cause unit test execution to fail. The reason I encountered unit test failure is that I did not use TestSidebarPart , but directly use SidebarPart.  

https://github.com/microsoft/vscode/blob/89db3e46dd79c88d2f4f25940ff2dc3bc9932c86/src/vs/workbench/test/browser/workbenchTestServices.ts#L663

When I need to initialize SidebarPart, it will report an error because I haven't registered any other views at the time of unit testing.


Another reason is that other modules are fetched using optional chain processing, because there is no guarantee that there will be a default view container.

![image](https://github.com/microsoft/vscode/assets/11473889/868306ea-6247-4fc9-8af0-57bcedc2fc3f)